src/sprightly.test.ts

```diff
@@ -0,0 +1,136 @@
+import fs from "fs"
+import { sprightlyAsync } from "./sprightly"
+
+jest.mock("fs", () => {
+  const dir: Record<string, string> = {
+    "my-name": `my name is {{ myName }}.\n{{> ../his-name }}`,
+    "../his-name": `his name is {{ hisName }}`,
+    nested: "my name is {{ name.first }} {{ name.last }}",
+    "array-indexing": "my name is {{ name[0] }} {{ name[1] }}",
+    "nested-indexing": "{{ property.nested[0].anotherProperty[1] }}",
+    "not-found-referenced-component": "{{> ../not-found }}",
+  }
+
+  return {
+    readFileSync: jest.fn((filePath: string) => {
+      if (filePath in dir) {
+        return dir[filePath]
+      }
+      throw new Error("file not found")
+    }),
+    existsSync: (filePath: string) => filePath in dir,
+  }
+})
+
+jest.mock("path", () => {
+  return {
+    ...jest.requireActual("path"),
```

Comment:
This is very nitpicky, but we don't really need this currently.

you can just mock `resolve` function and we would be set

As I said, very nitpicky